R/simulationResultsToDataFrame.R

```diff
@@ -54,83 +54,50 @@ simulationResultsToDataFrame <- function(simulationResults,
   )
 
   # leave out time units and dimensions since it is not a path
+  # they will be added at a later stage
   df_meta <- dplyr::filter(df_meta, paths != "Time")
 
   # combine dataframe with simulated data and meta data
   df <- dplyr::left_join(df_data, df_meta, by = "paths")
 
-  # add time unit to the this combined dataframe
-  # the time dimension is "Time", which is redundant and is thus left out
+  # now add previously left out time meta data to the combined dataframe
   df <- dplyr::bind_cols(
     df,
-    data.frame("TimeUnit" = simList$metaData$Time$unit[[1]])
+    data.frame(
+      "TimeUnit" = simList$metaData$Time$unit[[1]],
+      "TimeDimension" = simList$metaData$Time$dimension[[1]]
+    )
   )
 
-  # return the combined dataframe
-  return(df)
-}
-
-# custom function to convert dataframe from wide to long format
-# simplified version of
-# https://github.com/easystats/datawizard/blob/master/R/data_reshape.R
-# IP is author on this package
-
-.convert_to_long <- function(data,
-                             cols = "all",
-                             names_to = "Name",
-                             values_to = "Value") {
-  if (is.character(cols) && length(cols) == 1) {
-    if (cols == "all") {
-      # If all, take all
-      cols <- names(data)
-    } else {
-      # Surely, a regex
-      cols <- grep(cols, names(data), value = TRUE)
-    }
+  # if molecular weight is not present, extract it and add as a new column
+  if (!"molWeight" %in% names(df)) {
```

Comment:
`molWeight` is never part of `SimulationResults`, so don't have to check this.

And this function belongs to "utilities-simulation-results", I think.